rpcs3/Emu/Cell/Modules/cellCamera.cpp

```diff
@@ -370,19 +622,47 @@ s32 cellCameraRead(s32 dev_num, vm::ptr<u32> frame_num, vm::ptr<u32> bytes_read)
 
 s32 cellCameraReadEx(s32 dev_num, vm::ptr<CellCameraReadEx> read)
 {
-	UNIMPLEMENTED_FUNC(cellCamera);
+	cellCamera.todo("cellCameraReadEx(dev_num=%d, read=0x%x)", dev_num, read);
+
+	const auto g_camera = fxm::get<camera_thread>();
+
+	if (!g_camera)
+	{
+		return CELL_CAMERA_ERROR_NOT_INIT;
+	}
+	{
+		semaphore_lock lock(g_camera->mutex);
+
+		read->timestamp = g_camera->timer.GetElapsedTimeInMicroSec();
+		read->frame = g_camera->frame_num;
+		read->bytesread = g_camera->is_streaming ?
+			get_video_buffer_size(g_camera->info) : 0;
+	}
+
+	auto shared_data = fxm::make_always<gem_camera_shared>();
```

Comment:
Can this be get_always instead to avoid creating a new one each time?

Reply:
yea, not sure what I was thinking